--- src/main.c	2017-04-27 21:10:21.213169887 -0700
+++ ../vim.js/src/main.c	2017-04-27 21:10:23.356503178 -0700
@@ -1077,6 +1077,9 @@
     }
 #endif
 
+    int last_w = -1;
+    int last_h = -1;
+
     clear_oparg(&oa);
     while (!cmdwin
 #ifdef FEAT_CMDWIN
@@ -1084,6 +1087,15 @@
 #endif
 	    )
     {
+        int w;
+        int h;
+        gui_mch_get_screen_dimensions(&w, &h);
+        if (w != last_w || h != last_h){
+            gui_resize_shell(w, h);
+            last_w = w;
+            last_h = h;
+        }
+
 	if (stuff_empty())
 	{
 	    did_check_timestamps = FALSE;
